mos-platform/nes-gtrom/_prg-rom-banked.ld

```diff
@@ -0,0 +1,259 @@
+__prg_rom_9_lma = 0x098000;
+__prg_rom_10_lma = 0x0a8000;
+__prg_rom_11_lma = 0x0b8000;
+__prg_rom_12_lma = 0x0c8000;
+__prg_rom_13_lma = 0x0d8000;
+__prg_rom_14_lma = 0x0e8000;
+__prg_rom_15_lma = 0x0f8000;
+
+__prg_rom_0_offset = 0x000000;
+__prg_rom_1_offset = 0x008000;
+__prg_rom_2_offset = 0x010000;
+__prg_rom_3_offset = 0x018000;
+__prg_rom_4_offset = 0x020000;
+__prg_rom_5_offset = 0x028000;
+__prg_rom_6_offset = 0x030000;
+__prg_rom_7_offset = 0x038000;
+__prg_rom_8_offset = 0x040000;
+__prg_rom_9_offset = 0x048000;
+__prg_rom_10_offset = 0x050000;
+__prg_rom_11_offset = 0x058000;
+__prg_rom_12_offset = 0x060000;
+__prg_rom_13_offset = 0x068000;
+__prg_rom_14_offset = 0x070000;
+__prg_rom_15_offset = 0x078000;
+
+MEMORY {
+  /* PRG-ROM LMAs. */
+  prg_rom_0 : ORIGIN = __prg_rom_0_lma, LENGTH = __prg_rom_size >= 32 ? 0x8000 - 0x6 : 0
+  prg_rom_1 : ORIGIN = __prg_rom_1_lma, LENGTH = __prg_rom_size >= 64 ? 0x8000 - 0x6 : 0
+  prg_rom_2 : ORIGIN = __prg_rom_2_lma, LENGTH = __prg_rom_size >= 128 ? 0x8000 - 0x6 : 0
+  prg_rom_3 : ORIGIN = __prg_rom_3_lma, LENGTH = __prg_rom_size >= 128 ? 0x8000 - 0x6 : 0
+  prg_rom_4 : ORIGIN = __prg_rom_4_lma, LENGTH = __prg_rom_size >= 256 ? 0x8000 - 0x6 : 0
+  prg_rom_5 : ORIGIN = __prg_rom_5_lma, LENGTH = __prg_rom_size >= 256 ? 0x8000 - 0x6 : 0
+  prg_rom_6 : ORIGIN = __prg_rom_6_lma, LENGTH = __prg_rom_size >= 256 ? 0x8000 - 0x6 : 0
+  prg_rom_7 : ORIGIN = __prg_rom_7_lma, LENGTH = __prg_rom_size >= 256 ? 0x8000 - 0x6 : 0
+  prg_rom_8 : ORIGIN = __prg_rom_8_lma, LENGTH = __prg_rom_size >= 512 ? 0x8000 - 0x6 : 0
+  prg_rom_9 : ORIGIN = __prg_rom_9_lma, LENGTH = __prg_rom_size >= 512 ? 0x8000 - 0x6 : 0
+  prg_rom_10 : ORIGIN = __prg_rom_10_lma, LENGTH = __prg_rom_size >= 512 ? 0x8000 - 0x6 : 0
+  prg_rom_11 : ORIGIN = __prg_rom_11_lma, LENGTH = __prg_rom_size >= 512 ? 0x8000 - 0x6 : 0
+  prg_rom_12 : ORIGIN = __prg_rom_12_lma, LENGTH = __prg_rom_size >= 512 ? 0x8000 - 0x6 : 0
+  prg_rom_13 : ORIGIN = __prg_rom_13_lma, LENGTH = __prg_rom_size >= 512 ? 0x8000 - 0x6 : 0
+  prg_rom_14 : ORIGIN = __prg_rom_14_lma, LENGTH = __prg_rom_size >= 512 ? 0x8000 - 0x6 : 0
+  prg_rom_15 : ORIGIN = __prg_rom_15_lma, LENGTH = __prg_rom_size >= 512 ? 0x8000 - 0x6 : 0
+  vectors : ORIGIN = 0x10000 - 0x6, LENGTH = 6
+
+  /* CHR-ROM LMAs. */
+  chr_rom_0   : ORIGIN = 0x01000000, LENGTH = __chr_rom_size >= 8    ? 0x2000 : 0
+  chr_rom_1   : ORIGIN = 0x01002000, LENGTH = __chr_rom_size >= 16   ? 0x2000 : 0
+}
+
+REGION_ALIAS("c_readonly", prg_rom_0)
+
+SECTIONS {
+  /* Define a fixed section at the beginning of PRG-ROM bank 0. */
+  .prg_rom_fixed : {
+    *(.prg_rom_fixed .prg_rom_fixed.*)
+    __prg_rom_fixed_end = .;
+    __prg_rom_fixed_size = __prg_rom_fixed_end - __prg_rom_0_lma;
+  } >prg_rom_0
+
+  /* Offset each non-fixed bank section's LMA by the fixed section. */
+  /* This is accounted for in the custom output format. */
+  .prg_rom_0 : {
+      /* Non-fixed bank */
+      . = __prg_rom_0_lma + __prg_rom_fixed_size;
```

Comment:
Does this work as an address set before the colon in the section description? That's generally more idiomatic.

Reply:
Using `AT()`? No, it appears that it does not work - the functions get placed at `__prg_rom_%d_lma`, ignoring the fixed bank offset.

Reply:
Hm? No, just as `.prg_rom_0 __prg_rom_0_lma + __prg_rom_fixed_size : { }`

Reply:
Yes, you're right. That works fine. I have also learned that I have been writing too many dots in my link scripts :-)